### DISCUSSION TITLE
MainButton click passes event with evenData undefined

### DISCUSSION THREAD

**Discussion by reloverevolt** [vkruglikov/react-telegram-web-app#62] [Q&A]
Hello,

I have just noticed that MainButton has broken or something, yesterday it was functioning as expected , but now on each click console just reports:
![Screenshot from 2023-12-22 15-23-34](https://github.com/vkruglikov/react-telegram-web-app/assets/71507867/f3488f98-1e6a-4808-a177-67eb08f58d14)

my component is:
`<MainButton text="Buy" onClick={handleMainButtonClick}/>`

handler:
`function handleMainButtonClick(): void {
        WebApp.onEvent("mainButtonClicked", function(){
            const orderBody = { 
              user: profile?.userId,
              items: [{product: selectedProduct["id"], quantity: 1}],
              currency: "rub",
              promocode: null, 
              discount_id: discounts ? discounts[0]["id"] : null  
            };
            const requestOptions = {
              method: 'POST',
              headers: { 'Content-Type': 'application/json' },
              body: JSON.stringify(orderBody)
            };
      
            fetch(`${import.meta.env.BASE_URL}ecommerce/api/orders/`, requestOptions)
              .then(response => response.json())
              .then(data => WebApp.sendData(JSON.stringify(data)));
      
          });
    }`

I have checked and didn't find any recent updates on the side of telegram or repo 
Anyone has an idea of what could be the issue?

**Comment by vkruglikov** [comment]
Hi! It is definitely a Telegram issue
